packages/donor/src/screens/bookDonation/BookDonationScreenContainer.tsx

```diff
@@ -34,22 +40,53 @@ export function BookDonationScreenContainer({
   }
 
   const onSlotSelected = (donationSlot: DonationSlotToBook) => {
-    appointmentToBookStore.setAppointmentToBook(donationSlot);
-    if (isLoggedIn) {
-      navigate(MainNavigationKeys.Questionnaire);
+    setSelectedSlot(donationSlot);
+    const currentDonor = appState.donor;
+    const timeDifference = currentDonor?.lastCompletedDonationTimeMillis
+      ? DateUtils.getNumberOfDaysBetweenDates(
+          donationSlot.donationStartTimeMillis,
+          currentDonor.lastCompletedDonationTimeMillis
+        )
+      : MIN_DAYS_BETWEEN_DONATIONS + 1;
+
+    if (timeDifference < MIN_DAYS_BETWEEN_DONATIONS) {
+      setShowWarningPopup(true);
     } else {
-      navigate(MainNavigationKeys.Register);
+      continueDonation();
     }
   };
 
+  const continueDonation = () => {
+    if (selectedSlot) {
+      appointmentToBookStore.setAppointmentToBook(selectedSlot);
+      if (isLoggedIn) {
+        navigate(MainNavigationKeys.Questionnaire);
+      } else {
+        navigate(MainNavigationKeys.Register);
```

Comment:
I think we should put the login check first. If not logged in then move to Register.

Only then check if there's a selected slot.